test/functional/JIT_Test/src/jit/test/recognizedMethod/TestMathUtils.java

```diff
@@ -0,0 +1,311 @@
+            {+0.0f, -0.0f, false},
+            {-0.0f, +0.0f, true},
+
+            {+0.0f, fpInf, true},
+            {fpInf, +0.0f, false},
+            {+0.0f, fnInf, false},
+            {fnInf, +0.0f, true},
+            {-0.0f, fpInf, true},
+            {fpInf, -0.0f, false},
+            {-0.0f, fnInf, false},
+            {fnInf, -0.0f, true},
+
+            {+0.0d, +0.0d, true},
+            {-0.0d, -0.0d, true},
+            {+0.0d, -0.0d, false},
+            {-0.0d, +0.0d, true},
+
+            {+0.0d, dpInf, true},
+            {dpInf, +0.0d, false},
+            {+0.0d, dnInf, false},
+            {dnInf, +0.0d, true},
+            {-0.0d, dpInf, true},
+            {dpInf, -0.0d, false},
+            {-0.0d, dnInf, false},
+            {dnInf, -0.0d, true}
+        };
+    }
+
+    @DataProvider(name="nanProvider")
+    public static Object[][] nanProvider(){
+        Object[][] constNanPairs = new Object[][] {
+            {fqNaN, fqNaN, fqNaNBits},
+            {fsNaN, fsNaN, fsNaNBits},
+            {fqNaN, fsNaN, fqNaNBits},
+            {fsNaN, fqNaN, fsNaNBits},
+
+            {+0.0f, fqNaN, fqNaNBits},
+            {fqNaN, +0.0f, fqNaNBits},
+            {+0.0f, fsNaN, fsNaNBits},
+            {fsNaN, +0.0f, fsNaNBits},
+            {-0.0f, fqNaN, fqNaNBits},
+            {fqNaN, -0.0f, fqNaNBits},
+            {-0.0f, fsNaN, fsNaNBits},
+            {fsNaN, -0.0f, fsNaNBits},
+
+            {dqNaN, dqNaN, dqNaNBits},
+            {dsNaN, dsNaN, dsNaNBits},
+            {dqNaN, dsNaN, dqNaNBits},
+            {dsNaN, dqNaN, dsNaNBits},
+
+            {+0.0d, dqNaN, dqNaNBits},
+            {dqNaN, +0.0d, dqNaNBits},
+            {+0.0d, dsNaN, dsNaNBits},
+            {dsNaN, +0.0d, dsNaNBits},
+            {-0.0d, dqNaN, dqNaNBits},
+            {dqNaN, -0.0d, dqNaNBits},
+            {-0.0d, dsNaN, dsNaNBits},
+            {dsNaN, -0.0d, dsNaNBits}
+        };
+
+        Object[][] nanPairs = new Object[60][3];
+        int i = 0;
+        while (i < constNanPairs.length) {
+            nanPairs[i] = constNanPairs[i];
+            i++;
+        }
+        while (i < nanPairs.length) {
+            NaNTestPair<Float, Integer> fpair =  getFloatNaNPair();
+            NaNTestPair<Double, Long> dpair =  getDoubleNaNPair();
+            if (fpair == null) {
+                continue;
+            }
+            nanPairs[i][0] = fpair.first;
+            nanPairs[i][1] = fpair.second;
+            nanPairs[i][2] = fpair.expected;
+            i++;
+            if (dpair == null | i >= nanPairs.length){
+                continue;
+            }
+            nanPairs[i][0] = dpair.first;
+            nanPairs[i][1] = dpair.second;
+            nanPairs[i][2] = dpair.expected;
+            i++;
+        }
+        return nanPairs;
+    }
+
+    @DataProvider(name="normalNumberProvider")
+    public static Object[][] normalNumberProvider() {
+        Object [][] numList = new Object[20][2];
+        Random r = new Random();
```

Comment:
Instead of using a local new Random object in each method, can we have a static final Random object?